parallel_process_data.py

```diff
@@ -0,0 +1,600 @@
+
+    Args:
+        vals (list): Series of floats.
+        scores (tuple of float, optional): Percentile scores. Defaults to [5., 32., 50.].
+
+    Returns:
+        list of float: List with percentile ranks.
+
+    """
+    ranks = [percentileofscore(vals, s) for s in scores]
+    return ranks
+
+
+def calc_power(v, rho):
+    """"Determine power density.
+
+    Args:
+        v (float): Wind speed.
+        rho (float): Air density.
+
+    Returns:
+        float: Power density.
+
+    """
+    return .5 * rho * v ** 3
+
+
+def read_raw_data(start_year, final_year):
+    """"Read ERA5 wind data for adjacent years.
+
+    Args:
+        start_year (int): Read data starting from this year.
+        final_year (int): Read data up to this year.
+
+    Returns:
+        tuple of Dataset, ndarray, ndarray, ndarray, and ndarray: Tuple containing reading object of multiple wind
+        data (netCDF) files, longitudes of grid, latitudes of grid, model level numbers, and timestamps in hours since
+        1900-01-01 00:00:0.0.
+
+    """
+    # Construct the list of input NetCDF files
+    ml_files = []
+    sfc_files = []
+    for y in range(start_year, final_year+1):
+        for m in range(1, 13):
+            ml_files.append(path_join(era5_data_dir, model_level_file_name_format.format(y, m)))
+            sfc_files.append(path_join(era5_data_dir, surface_file_name_format.format(y, m)))
+    # Load the data from the NetCDF files.
+    ds = xr.open_mfdataset(ml_files+sfc_files, decode_times=False)
+
+    lons = ds['longitude'].values
+    lats = ds['latitude'].values
+
+    levels = ds['level'].values  # Model level numbers.
+    hours = ds['time'].values  # Hours since 1900-01-01 00:00:0.0, see: print(nc.variables['time']).
+
+    dlevels = np.diff(levels)
+    if not (np.all(dlevels == 1) and levels[-1] == 137):
+        i_highest_level = len(levels) - np.argmax(dlevels[::-1] > 1) - 1
+        print("Not all the downloaded model levels are consecutive. Only model levels up to {} are evaluated."
+              .format(levels[i_highest_level]))
+        levels = levels[i_highest_level:]
+    else:
+        i_highest_level = 0
+
+    return ds, lons, lats, levels, hours, i_highest_level
+
+
+def check_for_missing_data(hours):
+    """"Print message if hours are missing in timestamp series.
+
+    Args:
+        hours (list): Hour timestamps.
+
+    """
+    d_hours = np.diff(hours)
+    gaps = d_hours != 1
+    if np.any(gaps):
+        i_gaps = np.argwhere(gaps)
+        for i in i_gaps:
+            print("Gap found between {} and {}.".format(hour_to_date_str(hours[i]),
+                                                        hour_to_date_str(hours[i+1])))
+
+
+
+def process_lats_subset(lats_subset, lenLons, levels, lenHours, v_levels_east, v_levels_north, v_levels, t_levels, q_levels, surface_pressure, heights_of_interest, analyzed_heights_ids, res, start_time, counter, total_iters):
```

Comment:
please rewrite such that each parallel process outputs a separate netcdf, which we concatenate in a post processing step

Reply:
This is already happening, if read_n_lats_at_once is 1 - I had though to keep that variable, but at the moment if its larger, files span multiple latitudes - Now I removed the dependency on read_n_lats_at_once and now just process all latitudes individually 
54637c5 & 5dd45a1
Not sure if that's what you wanted though, or if its that best solution - so that we could talk about in Sykpe then.